cosmo_tester/test_suites/bootstrap_based_tests/sanity_scenario_test.py

```diff
@@ -0,0 +1,102 @@
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+import time
+import pytest
+
+from cosmo_tester.framework.test_hosts import BootstrapBasedCloudifyManagers
+
+from cosmo_tester.test_suites.snapshots import (
+    create_snapshot,
+    download_snapshot,
+    upload_snapshot,
+    restore_snapshot,
+    upgrade_agents,
+    delete_manager
+)
+
+from cosmo_tester.test_suites.ha.ha_helper import HighAvailabilityHelper \
+    as ha_helper
+
+
+@pytest.fixture(scope='module')
+def managers(cfy, ssh_key, module_tmpdir, attributes, logger):
+    """Bootstraps 3 Cloudify managers on a VM in Rackspace OpenStack."""
+
+    hosts = BootstrapBasedCloudifyManagers(
+        cfy, ssh_key, module_tmpdir, attributes, logger,
+        number_of_instances=3)
+
+    try:
+        hosts.create()
+        yield hosts.instances
+    finally:
+        hosts.destroy()
+
+
+def test_sanity_scenario(managers,
+                         cfy,
+                         hello_worlds,
+                         logger,
+                         tmpdir,
+                         attributes):
+    manager1 = managers[0]
+    manager3 = managers[2]
+    snapshot_id = 'SNAPSHOT_ID'
+    local_snapshot_path = str(tmpdir / 'snap.zip')
+
+    for hello in hello_worlds:
+        hello.upload_and_verify_install()
+
+    logger.info('Creating snapshot')
+    create_snapshot(manager1, snapshot_id, attributes, logger)
+    download_snapshot(manager1, local_snapshot_path, snapshot_id, logger)
+
+    manager3.use()
+
+    logger.info('Uploading and restoring snapshot')
+    upload_snapshot(manager3, local_snapshot_path, snapshot_id, logger)
+    restore_snapshot(manager3, snapshot_id, cfy, logger)
+
+    # wait a while to allow the restore-snapshot post-workflow commands to run
+    time.sleep(30)
+
+    logger.info('Upgrading agents')
+    upgrade_agents(cfy, manager3, logger)
+    delete_manager(manager1, logger)
+
+    for hello in hello_worlds:
+        hello.manager = manager3
+        hello.uninstall()
+        hello.delete_deployment()
+
+
+def _create_ha_cluster(cfy):
+    manager1 = managers.instances[0]
+    manager2 = managers.instances[1]
+
+    ha_helper.delete_active_profile()
+    manager1.use()
+    cfy.cluster.start(timeout=600,
+                      cluster_host_ip=manager1.private_ip_address,
+                      cluster_node_name=manager1.ip_address)
+
+    manager2.use()
+    cfy.cluster.join(manager1.ip_address,
+                     timeout=600,
+                     cluster_host_ip=manager2.private_ip_address,
+                     cluster_node_name=manager2.ip_address)
+
+    cfy.cluster.nodes.list()
```

Comment:
after you create the cluster best wait for it to stabilize before returning (join finish)

like this https://github.com/cloudify-cosmo/cloudify-system-tests/pull/646/files#diff-413e5570ec61db36da808db62184dc15R61